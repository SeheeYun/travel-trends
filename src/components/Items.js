import React from 'react';
import Item from './Item';

const Items = ({ items }) => {
  return (
    <div>
      {items.length > 0 &&
        items.map(item => <Item key={item.contentid} item={item} />)}
    </div>
  );
};

export default Items;
