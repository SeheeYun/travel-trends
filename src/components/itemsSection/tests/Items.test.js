import { render } from '@testing-library/react';
import Items from '../Items';

describe('Items', () => {
  const items = [
    {
      contentid: 123456,
      title: '에버랜드',
      firstimage: 'https://firstimage.jpg',
      firstimage2: 'https://firstimage2.jpg',
    },
    {
      contentid: 234567,
      title: '롯데월드',
      firstimage: 'https://firstimage3.jpg',
      firstimage2: 'https://firstimage4.jpg',
    },
  ];

  it('renders correctly with items', () => {
    const { container } = render(<Items items={items} />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly isLoading', () => {
    const { container } = render(<Items isLoading={true} />);
    expect(container).toMatchSnapshot();
  });
});
