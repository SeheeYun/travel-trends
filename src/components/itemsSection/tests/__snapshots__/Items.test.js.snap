// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items renders correctly isLoading 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  >
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`Items renders correctly with items 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  >
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <button
        class="MuiButtonBase-root"
        tabindex="0"
        type="button"
      >
        <a
          href="/view/123456"
        >
          <p>
            에버랜드
          </p>
          <div
            class="makeStyles-div-2"
          />
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="에버랜드"
              class="makeStyles-image-3"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://firstimage2.jpg);"
            />
            <noscript />
          </span>
        </a>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <button
        class="MuiButtonBase-root"
        tabindex="0"
        type="button"
      >
        <a
          href="/view/234567"
        >
          <p>
            롯데월드
          </p>
          <div
            class="makeStyles-div-2"
          />
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="롯데월드"
              class="makeStyles-image-3"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://firstimage4.jpg);"
            />
            <noscript />
          </span>
        </a>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Items renders correctly without items 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  >
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root makeStyles-root-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div />
    </div>
  </div>
</div>
`;
